frontend/src/metabase-types/api/mocks/card.ts

```diff
@@ -14,25 +14,30 @@ import {
   createMockNativeDatasetQuery,
   createMockStructuredDatasetQuery,
 } from "./query";
+import { createMockCollection } from "./collection";
 
-export const createMockCard = (opts?: Partial<Card>): Card => ({
-  id: 1,
-  name: "Question",
-  description: null,
-  display: "table",
-  public_uuid: null,
-  dataset_query: createMockStructuredDatasetQuery(),
-  visualization_settings: createMockVisualizationSettings(),
-  result_metadata: [],
-  dataset: false,
-  can_write: true,
-  cache_ttl: null,
-  collection_id: null,
-  last_query_start: null,
-  average_query_time: null,
-  archived: false,
-  ...opts,
-});
+export const createMockCard = (opts?: Partial<Card>): Card => {
+  const collection = createMockCollection();
+  return {
+    id: 1,
+    name: "Question",
+    description: null,
+    display: "table",
+    public_uuid: null,
+    dataset_query: createMockStructuredDatasetQuery(),
+    visualization_settings: createMockVisualizationSettings(),
+    result_metadata: [],
+    dataset: false,
+    can_write: true,
+    cache_ttl: null,
+    collection,
+    collection_id: collection.id as number,
```

Comment:
```suggestion
    collection_id: collection.id as CollectionId,
```

Reply:
Unfortunately, some endpoint doesn't always return `CollectionId`. This is [an example](https://github.com/metabase/metabase/pull/34827#issuecomment-1772602403), where instead of returning `collection_id: "root"`, BE just don't send `collection` or `collection_id` key at all. So the type for this property is actually `number | null`

https://github.com/metabase/metabase/blob/cef520443a69620698979cad3062020e10d182ff/frontend/src/metabase-types/api/card.ts#L17

Reply:
By the way, we agreed that [we won't change BE for that](https://github.com/metabase/metabase/pull/34827#issuecomment-1776562946) (won't send `collection: <root_collection>` and `collection_id: "root"`) so the current BE behavior will be as is.

Reply:
Ah ok, understandable. Then ignore my other comment about it 😀